.github/workflows/device-tests-android.yml

```diff
@@ -80,21 +76,68 @@ jobs:
 
       - name: AVD cache
         uses: actions/cache@v4
-        id: avd-cache
+        id: avd-cache-1
+        with:
+          path: |
+            ~/.android/avd/*
+            ~/.android/adb*
+          key: ${{ runner.os }}-avd-api${{ env.API-LEVEL-MIN }}
+
+      - name: Create AVD and generate snapshot for caching
+        if: steps.avd-cache-1.outputs.cache-hit != 'true'
+        timeout-minutes: 30
+        uses: reactivecircus/android-emulator-runner@6b0df4b0efb23bb0ec63d881db79aefbc976e4b2 # pin@v3
+        with:
+          api-level: ${{ env.API-LEVEL-MIN }}
+          # We don't need the Google APIs, but the default images are not available for 32+
+          target: ${{ env.API-LEVEL-MIN >= 32 && 'google_apis' || 'default' }}
+          force-avd-creation: false
+          ram-size: 2048M
+          arch: x86_64
+          disk-size: 4096M
+          emulator-options: -no-window -accel on -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          disable-animations: false
+          script: echo "Generated AVD snapshot for caching."
+
+      - name: Run Tests
+        timeout-minutes: 30
+        uses: reactivecircus/android-emulator-runner@6b0df4b0efb23bb0ec63d881db79aefbc976e4b2 # pin@v3
+        with:
+          api-level: ${{ env.API-LEVEL-MIN }}
```

Comment:
We'll run it for 2 API levels in sequence now? That'll slow down a lot the build. 

Are you sure it's due to parallelism? I don't get it why this would be

Reply:
Makes zero sense to me either. strategy-based jobs are completely different VMs, they don't influence one another in any implicit way.

Reply:
Sorry I didn't click the link. if it's due to caching, have you tried disabling it? Does it actually do much in terms of job runtime?

Reply:
Doing it right now.

Reply:
Sorry, this should have been a draft. I was trying to compare disabling AVD caching and putting them in sequence. And now I'm not sure what's going on. They run just as fast without caching?

Reply:
Alright, the 27 like to take its time.

Reply:
Let's go with disabling caching (takes 2 minutes longer) and we'll track enabling it in a separate issue